fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PruneAggChildColumns.java

```diff
@@ -54,15 +59,63 @@ public class PruneAggChildColumns extends OneRewriteRuleFactory {
     @Override
     public Rule build() {
         return RuleType.COLUMN_PRUNE_AGGREGATION_CHILD.build(logicalAggregate().then(agg -> {
+            Slot slot = handleAggregateConstant(agg);
+            List<Slot> childOutput = agg.child().getOutput();
+            if (slot != null) {
+                if (childOutput.size() == 1 && childOutput.get(0).equals(slot)) {
+                    return agg;
+                }
+                return agg.withChildren(ImmutableList.of(new LogicalProject<>(ImmutableList.of(slot), agg.child())));
+            }
             List<Expression> slots = Lists.newArrayList();
             slots.addAll(agg.getExpressions());
             Set<Slot> outputs = SlotExtractor.extractSlot(slots);
-            List<NamedExpression> prunedOutputs = agg.child().getOutput().stream().filter(outputs::contains)
+            List<NamedExpression> prunedOutputs = childOutput.stream().filter(outputs::contains)
                     .collect(Collectors.toList());
             if (prunedOutputs.size() == agg.child().getOutput().size()) {
                 return agg;
             }
             return agg.withChildren(ImmutableList.of(new LogicalProject<>(prunedOutputs, agg.child())));
         }));
     }
+
+    /**
+     * For these aggregate function with constant param. Such as:
+     *  count(*), count(1), sum(1)..etc.
+     */
+    private Slot handleAggregateConstant(LogicalAggregate<GroupPlan> agg) {
+        List<NamedExpression> outputExpressions = agg.getOutputExpressions();
+        for (NamedExpression namedExpression : outputExpressions) {
+            if (!(namedExpression instanceof Alias)) {
+                return null;
+            }
+            Expression childOfAlias = ((Alias) namedExpression).child();
+            if (!(childOfAlias instanceof AggregateFunction)) {
+                return null;
+            }
+            if (childOfAlias instanceof Count) {
```

Comment:
not just count need to handle. u need to check whether agg function's children contain slot reference.

Reply:
other agg function is processed in another branch.